ravenpy/models/emulators/canadianshield.py

```diff
@@ -0,0 +1,616 @@
+        :OW_Evaporation        {ow_evaporation}
+        :SWCanopyCorrect       SW_CANOPY_CORR_STATIC
+        :RainSnowFraction      {rain_snow_fraction}
+        :PotentialMeltMethod   POTMELT_DEGREE_DAY
+        :PrecipIceptFract      PRECIP_ICEPT_LAI
+        :SoilModel             SOIL_MULTILAYER 3
+
+        :MonthlyInterpolationMethod MONTHINT_LINEAR_MID
+
+        :Alias TOPSOIL          SOIL[0]
+        :Alias VADOSE           SOIL[1]
+        :Alias FRACBEDROCK      SOIL[2]
+
+        :HydrologicProcesses
+            :SnowRefreeze             FREEZE_DEGREE_DAY    SNOW_LIQ        SNOW
+            :Precipitation            PRECIP_RAVEN         ATMOS_PRECIP    MULTIPLE
+            :CanopyEvaporation        CANEVP_MAXIMUM       CANOPY          ATMOSPHERE
+            :CanopySnowEvap           CANEVP_MAXIMUM       CANOPY_SNOW     ATMOSPHERE
+            :SnowBalance              SNOBAL_TWO_LAYER     MULTIPLE        MULTIPLE
+            :Abstraction              ABST_FILL            PONDED_WATER    DEPRESSION
+            :OpenWaterEvaporation     OPEN_WATER_EVAP      DEPRESSION      ATMOSPHERE
+            :Infiltration             INF_HBV              PONDED_WATER    MULTIPLE
+            :Interflow                INTERFLOW_PRMS       TOPSOIL         SURFACE_WATER
+            :Baseflow                 BASE_POWER_LAW       VADOSE          SURFACE_WATER
+                  :-->Conditional HRU_TYPE IS SOIL_ORG
+            :Baseflow                 BASE_POWER_LAW       FRACBEDROCK     SURFACE_WATER
+                :Percolation          PERC_GAWSER          TOPSOIL         VADOSE
+                  :-->Conditional HRU_TYPE IS SOIL_ORG
+            :Percolation              PERC_GAWSER          VADOSE          FRACBEDROCK
+                  :-->Conditional HRU_TYPE IS SOIL_ORG
+                :Percolation          PERC_GAWSER          TOPSOIL         FRACBEDROCK
+                  :-->Conditional HRU_TYPE IS SOIL_BEDROCK
+            :SoilEvaporation          SOILEVAP_ROOT        TOPSOIL         ATMOSPHERE
+        :EndHydrologicProcesses
+        """
+
+        self.config.rvi.set_tmpl(rvi_tmpl)
+
+        self.config.rvi.routing = RVI.RoutingOptions.NONE
+        self.config.rvi.rain_snow_fraction = RVI.RainSnowFractionOptions.DINGMAN
+        self.config.rvi.evaporation = RVI.EvaporationOptions.PET_HARGREAVES_1985
+        self.config.rvi.ow_evaporation = RVI.EvaporationOptions.PET_HARGREAVES_1985
+
+    def derived_parameters(self):
+        params = cast(CANADIANSHIELD.Params, self.config.rvp.params)
+        self.config.rvp.derived_params = CANADIANSHIELD.DerivedParams(
+            SUM_X05_X06=params.par_x06,
+            SUM_X16_X17=params.par_x17,
+            POW_X08=params.par_x08,
+            POW_X09=params.par_x09,
+        )
+
+        # We must have two HRUs with equal areas
+        assert len(self.config.rvh.hrus) == 2
+        assert self.config.rvh.hrus[0].area == self.config.rvh.hrus[1].area
```

Comment:
Replace by try/expect statements. assert statements are removed with the -O optimization flag.